src/nodevisitor.rs

```diff
@@ -70,7 +70,7 @@ impl NodeVisitor for StaticPathParser {
 
                 (Some(ParseToken::Number(num)), ParseToken::ArrayEof) => {
                     self.static_path_elements
-                        .push(StaticPathElement::ArrayIndex(num.clone()));
+                        .push(StaticPathElement::ArrayIndex(*num));
```

Comment:
Yeah, this is cleaner. How did you find this?

Reply:
`cargo clippy`